src/modules/dashboard/components/Extensions/ExtensionDetails.tsx

```diff
@@ -0,0 +1,338 @@
+                />
+              );
+            }}
+          />
+        </Switch>
+        <aside>
+          <div className={styles.buttonWrapper}>
+            <ExtensionActionButton
+              canInstall={canInstall}
+              colonyAddress={colonyAddress}
+              installedExtension={installedExtension}
+              extension={extension}
+            />
+          </div>
+          <Table appearance={{ theme: 'lined' }}>
+            <TableBody>
+              {tableData.map(({ label, value }) => (
+                <TableRow key={label.id}>
+                  <TableCell className={styles.cellLabel}>
+                    <FormattedMessage {...label} />
+                  </TableCell>
+                  <TableCell className={styles.cellData}>{value}</TableCell>
+                </TableRow>
+              ))}
+            </TableBody>
+          </Table>
+          {canInstall &&
+          extension.uninstallable &&
+          installedExtension &&
+          !installedExtension.details.deprecated ? (
+            <div className={styles.buttonUninstall}>
+              <DialogActionButton
+                dialog={ConfirmDialog}
+                dialogProps={{
+                  confirmButtonText: MSG.confirmDeprecate,
+                  children: <FormattedMessage {...MSG.textDeprecate} />,
+                }}
+                appearance={{ theme: 'blue' }}
+                submit={ActionTypes.COLONY_EXTENSION_DEPRECATE}
+                error={ActionTypes.COLONY_EXTENSION_DEPRECATE_ERROR}
+                success={ActionTypes.COLONY_EXTENSION_DEPRECATE_SUCCESS}
+                text={MSG.buttonUninstall}
+                values={{ colonyAddress, extensionId }}
+              />
+            </div>
+          ) : null}
+          {canInstall &&
+          extension.uninstallable &&
+          installedExtension &&
+          installedExtension.details.deprecated ? (
+            <div className={styles.buttonUninstall}>
+              <DialogActionButton
+                dialog={ConfirmDialog}
+                dialogProps={{
+                  confirmButtonText: MSG.confirmUninstall,
+                  children: <FormattedMessage {...MSG.textUninstall} />,
+                }}
+                appearance={{ theme: 'blue' }}
+                submit={ActionTypes.COLONY_EXTENSION_UNINSTALL}
+                error={ActionTypes.COLONY_EXTENSION_UNINSTALL_ERROR}
+                success={ActionTypes.COLONY_EXTENSION_UNINSTALL_SUCCESS}
+                text={MSG.buttonUninstall}
+                values={{ colonyAddress, extensionId }}
+              />
+            </div>
+          ) : null}
+          <div className={styles.permissions}>
+            <Heading
+              appearance={{ size: 'normal' }}
+              tagName="h4"
+              text={MSG.permissionsNeeded}
+            />
+            {extension.neededColonyPermissions.map((permission: ColonyRole) => (
+              <Tag
```

Comment:
Note that @alicjakujawa is reimplementing the proper version of these permissions tags in #2292 

Leave them as they are here, we'll deal with updating them once the above PR is merged